dandiapi/api/services/dandiset/__init__.py

```diff
@@ -74,3 +74,45 @@ def delete_dandiset(*, user, dandiset: Dandiset) -> None:
 
         dandiset.versions.all().delete()
         dandiset.delete()
+
+
+def star_dandiset(*, user, dandiset: Dandiset) -> int:
+    """
+    Star a Dandiset for a user.
+
+    Args:
+        user: The user starring the Dandiset.
+        dandiset: The Dandiset to star.
+
+    Returns:
+        The new star count for the Dandiset.
+    """
+    if not user.is_authenticated:
+        return dandiset.star_count
+
+    with transaction.atomic():
+        DandisetStar.objects.get_or_create(user=user, dandiset=dandiset)
+        # Refresh star count from database
+        dandiset.refresh_from_db()
+        return dandiset.star_count
```

Comment:
There's not really a reason to use `transaction.atomic` here, since there's only one write action being performed. The `refresh_from_db` call is also superfluous, since `star_count` makes a query to the DB anyway.

```suggestion
    DandisetStar.objects.get_or_create(user=user, dandiset=dandiset)
    return dandiset.star_count
```